src/puppetlabs/puppetdb/command.clj

```diff
@@ -473,7 +473,9 @@
   (-> cmdref
       (queue/cons-attempt ex)
       (dlo/discard-cmdref q dlo))
-  (mark-both-metrics! (:command cmdref) (:version cmdref) :discarded))
+  (mark-both-metrics! (:command cmdref) (:version cmdref) :discarded)
+  (let [{:keys [command version]} cmdref]
+    (update-counter! :depth command version dec!)))
```

Comment:
This looks good, but let's pull `mark-both-metrics!` into the `let`, so it can use the `command` and `version` bindings too.